### PR TITLE
[Merged by Bors] - chore: adapt to multiple goal linter 6

### PR DESCRIPTION
**Pull request by adomani**
Another halfway split of #12381 on the way to #12352.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/Perm/Cycle/Factors.lean [hunk 1]
```diff
@@ -607,7 +607,7 @@ theorem cycle_is_cycleOf {f c : Equiv.Perm α} {a : α} (ha : a ∈ c.support)
   let hfc := (Equiv.Perm.disjoint_mul_inv_of_mem_cycleFactorsFinset hc).symm
   let hfc2 := Perm.Disjoint.commute hfc
   rw [← Equiv.Perm.cycleOf_mul_of_apply_right_eq_self hfc2]
-  simp only [hfc2.eq, inv_mul_cancel_right]
+  · simp only [hfc2.eq, inv_mul_cancel_right]
```
Comment (grunweg) [inline Mathlib/GroupTheory/Perm/Cycle/Factors.lean:610]: I wouldn't mind the comment below this line getting translated (I speak some French, but not everybody does) - but this is pre-existing.

File: Mathlib/LinearAlgebra/Prod.lean [hunk 2]
```diff
@@ -1006,7 +1006,7 @@ theorem tailings_disjoint_tunnel (f : M × N →ₗ[R] M) (i : Injective f) (n :
     apply tailing_disjoint_tunnel_succ
   · simp only [tailings_succ]
     refine' Disjoint.disjoint_sup_left_of_disjoint_sup_right _ _
-    apply tailing_disjoint_tunnel_succ
+    · apply tailing_disjoint_tunnel_succ
```
Comment (grunweg) [inline Mathlib/LinearAlgebra/Prod.lean:1009]: For just reading the source, using `refine` would be quite clearer: AIUI, the `refine` creates two goals, the first of which is cdot-ed?

File: Mathlib/MeasureTheory/Covering/VitaliFamily.lean [hunk 3]
```diff
@@ -206,7 +206,7 @@ def enlarge (v : VitaliFamily μ) (δ : ℝ) (δpos : 0 < δ) : VitaliFamily μ
     have : ∀ x ∈ s, ∀ ε : ℝ, ε > 0 → ∃ (a : Set α), a ∈ g x ∧ a ⊆ closedBall x ε := by
       intro x hx ε εpos
       obtain ⟨a, af, ha⟩ : ∃ a ∈ f x, a ⊆ closedBall x (min ε δ)
-      exact ffine x hx (min ε δ) (lt_min εpos δpos)
+      · exact ffine x hx (min ε δ) (lt_min εpos δpos)
```
Comment (grunweg) [inline Mathlib/MeasureTheory/Covering/VitaliFamily.lean:209]: Can you use the `obtain ... := by exact ...` syntax here? (This instance was missed in #11045.)
Or would you prefer me filing a PR for this and few similar instances, which can land before this one?

File: Mathlib/MeasureTheory/Measure/Haar/Basic.lean [hunk 4]
```diff
@@ -254,16 +254,18 @@ theorem index_union_eq (K₁ K₂ : Compacts G) {V : Set G} (hV : (interior V).N
...
   apply Finset.disjoint_filter.mpr
   rintro g₁ _ ⟨g₂, h1g₂, h2g₂⟩ ⟨g₃, h1g₃, h2g₃⟩
   simp only [mem_preimage] at h1g₃ h1g₂
   refine' h.le_bot (_ : g₁⁻¹ ∈ _)
   constructor <;> simp only [Set.mem_inv, Set.mem_mul, exists_exists_and_eq_and, exists_and_left]
-  · refine' ⟨_, h2g₂, (g₁ * g₂)⁻¹, _, _⟩; simp only [inv_inv, h1g₂]
-    simp only [mul_inv_rev, mul_inv_cancel_left]
-  · refine' ⟨_, h2g₃, (g₁ * g₃)⁻¹, _, _⟩; simp only [inv_inv, h1g₃]
-    simp only [mul_inv_rev, mul_inv_cancel_left]
+  · refine' ⟨_, h2g₂, (g₁ * g₂)⁻¹, _, _⟩
```
Comment (grunweg) [inline Mathlib/MeasureTheory/Measure/Haar/Basic.lean:263]: I know you wanted to make minimal changes, but... I think that, like above, `refine` would make the source slightly clearer.

Would you prefer a separate PR with just this clean-up?

File: Mathlib/MeasureTheory/Measure/Typeclasses.lean [hunk 5]
```diff
@@ -1555,7 +1555,7 @@ noncomputable irreducible_def MeasureTheory.Measure.finiteSpanningSetsInOpen' [T
     rw [not_nonempty_iff_eq_empty.1 h'T, sUnion_empty] at hT
     simpa only [← hT] using mem_univ (default : α)
   obtain ⟨f, hf⟩ : ∃ f : ℕ → Set α, T = range f
-  exact T_count.exists_eq_range T_ne
+  · exact T_count.exists_eq_range T_ne
```
Comment (grunweg) [inline Mathlib/MeasureTheory/Measure/Typeclasses.lean:1558]: Again, `obtain ... := by ...` seems cleaner here.

File: Mathlib/NumberTheory/Padics/Hensel.lean [hunk 6]
```diff
@@ -303,7 +303,7 @@ private theorem newton_seq_dist_aux (n : ℕ) :
   | k + 1 =>
     have : 2 ^ n ≤ 2 ^ (n + k) := by
       apply pow_le_pow_right
-      norm_num
+      · norm_num
```
Comment (grunweg) [inline Mathlib/NumberTheory/Padics/Hensel.lean:306]: I'm on the fence whether indenting both goals or just the first one is better. I don't strongly object to your current choice.

**Review by grunweg — APPROVED** [review]
Looks good to me: in a few places, I would tweak the code a little further (but these are minor comments).

**Comment by grunweg** [comment]
Filed #12850 and #12851 with the style tweaks I suggested. Reviews welcome - since *I* wrote that code, you should have no conflict of interest :-)

File: Mathlib/MeasureTheory/Covering/VitaliFamily.lean (hunk 3 above)
Comment (adomani) [reply Mathlib/MeasureTheory/Covering/VitaliFamily.lean:209]: Done!

File: Mathlib/MeasureTheory/Measure/Haar/Basic.lean (hunk 4 above)
Comment (adomani) [reply Mathlib/MeasureTheory/Measure/Haar/Basic.lean:263]: Done!

File: Mathlib/MeasureTheory/Measure/Typeclasses.lean (hunk 5 above)
Comment (adomani) [reply Mathlib/MeasureTheory/Measure/Typeclasses.lean:1558]: Done!

File: Mathlib/LinearAlgebra/Prod.lean (hunk 2 above)
Comment (adomani) [reply Mathlib/LinearAlgebra/Prod.lean:1009]: Done!  And indented the second one as well.

File: Mathlib/GroupTheory/Perm/Cycle/Factors.lean (hunk 1 above)
Comment (adomani) [reply Mathlib/GroupTheory/Perm/Cycle/Factors.lean:610]: Translation provided!

File: Mathlib/NumberTheory/Padics/Hensel.lean (hunk 6 above)
Comment (adomani) [reply Mathlib/NumberTheory/Padics/Hensel.lean:306]: I ended up indenting this one as well!

**Comment by adomani** [comment]
Michael, thank you very much for the continuing thorough reviews!

**Comment by grunweg** [comment]
Thank you for writing this linter, Damiano - I would very much like to see it land!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors merge

bors d+

**Pull request closed without merging**